src/testing/resolvers/zones.ts

```diff
@@ -0,0 +1,147 @@
+import { http, HttpResponse } from "msw";
+import { afterEach } from "vitest";
+
+import type {
+  ZoneRequest,
+  ZonesWithSummaryListResponse,
+} from "@/app/apiclient";
+import { zone as zoneFactory } from "@/testing/factories";
+import { BASE_URL } from "@/testing/utils";
+
+const initialMockZones: ZonesWithSummaryListResponse = {
+  items: [
+    zoneFactory({
+      id: 1,
+      name: "zone-1",
+      description: "",
+      controllers_count: 0,
+      devices_count: 0,
+      machines_count: 0,
+    }),
+    zoneFactory({
+      id: 2,
+      name: "zone-2",
+      description: "",
+      controllers_count: 0,
+      devices_count: 0,
+      machines_count: 0,
+    }),
+    zoneFactory({
+      id: 3,
+      name: "zone-3",
+      description: "",
+      controllers_count: 0,
+      devices_count: 0,
+      machines_count: 0,
+    }),
+  ],
+  total: 3,
+};
+
+let mockZones = structuredClone(initialMockZones);
+
+const isValidZoneRequest = (data: any): data is ZoneRequest =>
+  typeof data === "object" &&
+  data !== null &&
+  typeof data.name === "string" &&
+  (!data.description || typeof data.description === "string");
+
+const zoneResolvers = {
+  listZones: {
+    resolved: false,
+    handler: () =>
+      http.get(`${BASE_URL}MAAS/a/v3/zones_with_summary`, () => {
```

Comment:
I feel like we could maybe keep track of API endpoint URLs in a const or something, but this is beyond the scope here. I'll add this as a comment to [MAASENG-4386](https://warthogs.atlassian.net/browse/MAASENG-4386).